### PR TITLE
feat(x/ecocredit): query baskets

### PR DESCRIPTION
**Pull request by aaronc**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/basket/query_util.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+	to := &queryv1beta1.PageRequest{}
+	err := GogoToPulsarSlow(from, to)
+	return to, err
+}
+
+func PulsarPageResToGogoPageRes(from *queryv1beta1.PageResponse) (*query.PageResponse, error) {
+	if from == nil {
+		return nil, nil
+	}
```
Comment (ryanchristo) [inline x/ecocredit/server/basket/query_util.go:23]: Should we be returning an empty page response here?

**Review by ryanchristo — APPROVED** [review]
Looks good to me. One question.

File: x/ecocredit/server/basket/query_baskets.go [hunk 2]
```diff
@@ -3,10 +3,49 @@ package basket
...
+
+	res := &baskettypes.QueryBasketsResponse{}
+	for it.Next() {
+		basket, err := it.Value()
+		if err != nil {
+			return nil, err
+		}
+
+		basketGogo := &baskettypes.Basket{}
+		err = PulsarToGogoSlow(basket, basketGogo)
```
Comment (technicallyty) [inline x/ecocredit/server/basket/query_baskets.go:39]: would it not be more efficient to just set the fields directly on basketGogo? i.e.

```go
basketGogo := &baskettypes.Basket{
    Id: basket.Id,
    // ...
}
```

Comment (aaronc) [reply x/ecocredit/server/basket/query_baskets.go:39]: Yes it would. It would also be more brittle because what if a field gets added?

Comment (technicallyty) [reply x/ecocredit/server/basket/query_baskets.go:39]: makes sense, suppose its only a bandaid till we can get rid of gogo fully anyways

**Review by technicallyty — APPROVED** [review]

File: x/ecocredit/server/basket/query_util.go (hunk 1 above)
Comment (aaronc) [reply x/ecocredit/server/basket/query_util.go:23]: Well it's not really necessary. This is just a bandaid anyway

**Pull request merged**